src/auto-complete/src/AutoComplete.tsx

```diff
@@ -238,14 +238,11 @@ export default defineComponent({
     function select (option: AutoCompleteOption): void {
       if (option?.value !== undefined) {
         doSelect(option.value)
+        doUpdateValue(
+          props.append ? `${mergedValueRef.value}${option.value}` : option.value
+        )
         if (props.clearAfterSelect) {
           doUpdateValue(null)
-        } else if (option.label !== undefined) {
-          doUpdateValue(
-            props.append
-              ? `${mergedValueRef.value}${option.label}`
-              : option.label
-          )
         }
```

Comment:
上方的doUpdateValue放到if的第二分支判断里会更好点？
```suggestion
} else if(props.append){
     doUpdateValue(`${mergedValueRef.value}${option.value}`)
}
```

Reply:
> 上方的doUpdateValue放到if的第二分支判断里会更好点？

append 为 false 也需要 doUpdateValue

Reply:
> > 上方的doUpdateValue放到if的第二分支判断里会更好点？
> 
> append 为 false 也需要 doUpdateValue

那应该放到else里，否则会在clearAfterSelect为true时重复执行doUpdateValue
```suggestion
        } else{
        doUpdateValue(
          props.append ? `${mergedValueRef.value}${option.value}` : option.value
        )
        }
```

Reply:
> > > 上方的doUpdateValue放到if的第二分支判断里会更好点？
> > 
> > 
> > append 为 false 也需要 doUpdateValue
> 
> 那应该放到else里，否则会在clearAfterSelect为true时重复执行doUpdateValue

是的